controllers/diet_requirement_retrieval.py

```diff
@@ -0,0 +1,59 @@
+from flask import Blueprint
+from flask_restful import reqparse, fields, Resource, marshal_with, Api
+from flask import request
+from domain import session_scope
+
+from repository.diet_requirement_repository import get_dietary_requirements
+
+
+get_parser = reqparse.RequestParser()
+get_parser.add_argument('user_id', type=int, required=True)
+
+
+diet_requirement_fields = {
+    'diet_id': fields.Integer,
+    'user_id': fields.Integer,
+    'halal': fields.Boolean,
+    'vegetarian': fields.Boolean,
+    'vegan': fields.Boolean,
+    'nut_allergy': fields.Boolean,
+    'shellfish_allergy': fields.Boolean,
+    'gluten_intolerance': fields.Boolean,
+    'kosher': fields.Boolean,
+    'lactose_intolerance': fields.Boolean,
+    'diabetic': fields.Boolean,
+    'egg_allergy': fields.Boolean,
+    'other': fields.String
+}
+
+
+result_fields = {
+    "diet_requirement": fields.Nested(diet_requirement_fields),
+    "success": fields.Boolean
+}
+
+class RetrieveDietaryRequirement(Resource):
+    """
+    This is a class to retrieve the dietary requirement data from the database
+    """
+
+    @marshal_with(result_fields)
+    def get(self):
+        try:
+            print("RetrieveDietaryRequirement.get() called")
```

Comment:
We should try to remove print statements before pushing to main, unless they are still needed for something. In this case, I don't think the print statement is still necessary so it should be removed.